### ISSUE TITLE
XdocsExamplesAstConsistencyTest`: false positive "Comments mismatch" in `extractComments`

### ISSUE THREAD

**Issue by smita1078** [checkstyle/checkstyle#20307]
## Summary

`testExamplesDifferOnlyByComments` reports false positive _Comments mismatch_ violations for example pairs that are structurally equivalent. The root cause is in `extractComments`: it fails to skip three categories of documentation marker comments, causing the comment lists of two otherwise-matching examples to differ.

## Observed failures (sample)

```
Directory: checks/whitespace/whitespacearound
Reference: Example1.java
Mismatch:  Example3.java
Reason:    Comments mismatch

Directory: checks/whitespace/whitespacearound
Reference: Example1.java
Mismatch:  Example11.java
Reason:    Comments mismatch
```

---

### 1. `// N violations` is not skipped

`extractComments` skips comments starting with `"violation"`, but `"3 violations"` starts with a digit so it is **kept**. Meanwhile the equivalent line in Example3 uses `// violation '...'` which **is** skipped — the extracted lists are asymmetric.

```java
// Example1.java
public Example1(){} // 3 violations          ← starts with "3" → KEPT (bug)

// Example3.java
public Example3(){} // violation '...'       ← starts with "violation" → skipped ✓
```

Result: Example1 list = `["3 violations", ...]`, Example3 list = `[]` → false mismatch.

---

### 2. Continuation explanation lines are not skipped

Example1 follows count-style markers with a standalone explanation on the next line. That line does not match any skipped prefix so it is **kept**:

```java
// Example1.java
catch (Exception e){} // 3 violations
// no space after ')' and '{', no space before '}'   ← kept → causes mismatch
```

Example3 has no matching continuation line, so the lists still differ even after fixing 1.

---

### 3. `// filtered violation` is not skipped

`"filtered violation".startsWith("violation")` is `false` — it starts with `"filtered"`. Comments like:

```java
// filtered violation above ''42' is a magic number'
```

are kept in the comparison list when they should be skipped.

**Comment by smita1078** [comment]
@stoyanK7 @romani Came across this while fixing other examples, hence created this issue

**Comment by stoyanK7** [comment]
@smita1078 Please feel free to proceed with fixes.

**Comment by stoyanK7** [comment]
Fix is merged
